### DISCUSSION TITLE
Trigger manual re-query

### DISCUSSION THREAD

**Discussion by arcanis** [urql-graphql/urql#1782] [Q&A] [answered]
I've seen that you advocate for invalidating the cache from the mutations, which I think is the right strategy in general. However, I'm facing a case a bit special where I believe the invalidation is really meant to be triggered by a user interaction, not by a mutation - in my case, it would be refreshing the data when the user switches back to the page tab (using the `focus` event from `document`). Is there a clean way to achieve this?

**Comment by JoviDeCroock** [comment, marked as answer]
Would something like the [`refocus-exchange`](https://github.com/FormidableLabs/urql/tree/main/exchanges/refocus) be sufficient?

**Reply by arcanis** [reply]
Oh! I don't think this existed when I discovered urql! That's exactly what I needed, thanks! 🌟

(slight off topic, it's also quite nice how small is its code, it's handy if one needs to build a similar custom behavior later)
